### PR TITLE
 Make it unnecessary to append a '_' to the controller name when using the AdminSecurity annotation

### PR DESCRIPTION
**Pull request by eternoendless**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Read below
| Type?         | improvement
| Category?     | CO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | n/a
| How to test?  | Access any of the migrated pages. If you can, it means the change works.

Currently, if you use the `@AdminSecurity` annotation in "modern" controllers, you need to append an underscore to the legacy controller name, like this:

```
request.get('_legacy_controller')~'_'
```
Let's take the following example from AdministrationController.php:

```php
/**
 * @AdminSecurity("is_granted('read', request.get('_legacy_controller')~'_')", message="Access denied.")
 */
```

Roles (permissions) for controllers are stored in the database as `ROLE_MOD_TAB_<CONTROLLERNAME>_<CREATE|READ|UPDATE|DELETE>`. 

In this example, the corresponding role is `ROLE_MOD_TAB_ADMINADMINPREFERENCES_READ`.

The `PageVoter` that is invoked when calling `is_granted()` receives "read" as attribute, and "AdminAdminPreferences" as subject (specified by the `_legacy_controller` property in the route yaml).

Using those elements, it builds the role ID and then hands it over to the legacy access control which performs the check.

Until now, the role was created by joining the subject and the attribute together, then converting them to upper case. This PR adds an underscore between them if there isn't one there already (to ensure backwards compatibility).

That's all! Now we can have cleaner calls to AdminSecurity. Check out said change in this commit: https://github.com/PrestaShop/PrestaShop/commit/95ce71caa181e07c0bd53d1fb04a42ddc0bfe3cf

The second commit is where I changed all the current calls so that they conform to the new spec.

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/9432)

### CODE REVIEW

File: src/PrestaShopBundle/Security/Voter/PageVoter.php [hunk 1]
```diff
@@ -49,41 +52,63 @@ class PageVoter extends Voter
...
      * @return bool
      */
     protected function supports($attribute, $subject)
     {
-        if (!in_array($attribute, array(self::CREATE, self::UPDATE, self::DELETE, self::READ))) {
-            return false;
-        }
-
-        return true;
+        return in_array($attribute, array(self::CREATE, self::UPDATE, self::DELETE, self::READ));
```
Comment (PierreRambaud) [inline src/PrestaShopBundle/Security/Voter/PageVoter.php:63]: :+1:

**Review by PierreRambaud — APPROVED** [review]

**Review by mickaelandrieu — APPROVED** [review]

**Comment by mickaelandrieu** [comment]
would you mind to update the Symfony migration guide docs too? :)

**Comment by eternoendless** [comment]
@mickaelandrieu Yeah I'm on it :)

**Comment by mickaelandrieu** [comment]
Thanks everyone !

**Pull request merged**